jhub_apps/spawner/proxy/oauth2.py

```diff
@@ -0,0 +1,166 @@
+                return True
+
+            proc = SupervisedProcess(
+                key,
+                *cmd,
+                env=self.get_env(),
+                ready_func=ready_func,
+                ready_timeout=READY_TIMEOUT,
+                log=self.log,
+                stdout=subprocess.PIPE,
+                stderr=subprocess.PIPE,
+            )
+            self._procs[key] = proc
+
+            try:
+                await proc.start()
+                if not await proc.ready():
+                    self.log.error(f"{key} failed to become ready")
+                    out, err = await proc.proc.communicate()
+                    self.log.error(out.decode())
+                    self.log.error(err.decode())
+                    await proc.kill()
+                    del self._procs[key]
+                    return False
+
+                asyncio.ensure_future(self._drain_logs(proc, key))
+            except Exception:
+                del self._procs[key]
+                raise
+
+            return True
+
+    async def start(self) -> bool:
+        # Start main proxy process toghether with its target proxied applications
+        return NotImplemented
+
+
+    async def stop(self) -> bool:
+        # Stop all the processes started by this handler
+        # Added just in case, but not used in the current implementation
+        async with self._lock:
+            for key, proc in self._procs.items():
+                await proc.kill()
+                del self._procs[key]
+        return True
+
+
+class OAuth2ProxyHandler(ProxyHandler):
+    # This class is responsible for managing the oauth2-proxy process
+    # and its target proxied applications
+
+    def __init__(
+            self,
+            log: logging.Logger,
+            oauth2_proxy_cmd: List[str],
+            target_app_cmd: List[str],
+    ):
+        super().__init__(log)
+        self.oauth2_proxy_cmd = oauth2_proxy_cmd
+        self.target_app_cmd = target_app_cmd
+
+
+
+# @click.command()
+# @click.option('--destport', default=0, help='port that the webapp should end up running on; default 0 to be assigned a random free port')
+# @click.option('--ip', default=None, help='Address to listen on')
+# @click.option('--presentation-path', default='', help='presentation_path substitution variable')
+# @click.option('--debug/--no-debug', default=False, help='To display debug level logs')
+# @click.option('--logs/--no-logs', default=True, help='To display logs generated by the subprocess')
+# @click.option('--authtype', type=click.Choice(['oauth', 'none'], case_sensitive=True), default='oauth')
+# @click.option('--request-timeout', default=300, type=click.INT, help='timeout of proxy http calls to subprocess in seconds (default 300)')
+# @click.option('--force-alive/--no-force-alive', default=False, help='Force the process to stay alive even if it is not ready')
+# @click.option('--ready-check-path', default='/', help='URL path to poll for readiness (default /)')
```

Comment:
I will remove all this later on, just left here to keep some backward compatibility